### ISSUE TITLE
Use Set.of() instead of new HashSet<>(Arrays.asList()) in MetadataConstraints.

### ISSUE THREAD

**Issue by keith-turner** [apache/accumulo#2040]
Java recently introduced the Set.of() static methods for creating immutable sets.  These new methods could be used [here](https://github.com/apache/accumulo/blob/9afab2c1b931d100ed6bad5cfe028c28da944427/server/base/src/main/java/org/apache/accumulo/server/constraints/MetadataConstraints.java#L79) and [here](https://github.com/apache/accumulo/blob/9afab2c1b931d100ed6bad5cfe028c28da944427/server/base/src/main/java/org/apache/accumulo/server/constraints/MetadataConstraints.java#L90) instead of `new HashSet<>(Arrays.asList(`. This would shorten the code and make the sets immutable, which seems to be the intention of the code.

**Comment by keith-turner** [comment]
There are other places where this type of change could be made. In Linux can run the following command on the Accumulo source code to look possible places to make this change.

```bash
find . -name "*.java" | xargs grep asList | grep new | grep Set 
```

**Comment by hguercan** [comment]
Hi, I would like to work on this issue

**Comment by EdColeman** [comment]
Closed with #2065
